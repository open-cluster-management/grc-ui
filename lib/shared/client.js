/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

/*
 * A boolean module that is true on the client, false on the server. This is for code that needs to
 * do different things on the server vs the client.
 */

module.exports = (typeof window !== 'undefined')
