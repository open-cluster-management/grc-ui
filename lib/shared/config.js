/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
var requireServer = require('./require-server'),
    nconf = requireServer('nconf')

var WHITELIST = [
  'contextPath',
  'headerContextPath',
  'clusterContextPath',
  'grcUiApiUrl',
  'complianceNamespace',
  'overview',
  'featureFlags: filters',
  'featureFlags:dashboardLiveUpdates',
  'featureFlags:dashboardRefreshInterval',
  'featureFlags:fullDashboard',
  'featureFlags:liveUpdates',
  'featureFlags:liveUpdatesPollInterval',
  'featureFlags:search',
  'feature_searchRelated',
  'feature_search-api',
  'feature_security-findings',
  'feature_applications',
  'ICP_VERSION'
]

var config = {}

if (nconf) {
  WHITELIST.forEach(i => config[i] = nconf.get(i))
  config.env = process.env.NODE_ENV
} else {
  const configElement = document.getElementById('config')
  config = (configElement && JSON.parse(configElement.textContent)) || {}
}

module.exports = config
