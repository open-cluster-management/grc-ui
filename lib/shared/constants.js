/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

module.exports.ACM_ACCESS_COOKIE = 'acm-access-token-cookie'
module.exports.CFC_AUTH_COOKIE = 'cfc-acs-auth-cookie'

module.exports.REFRESH_TIMES = [5, 10, 30, 60, 5*60, 30*60, 0]
module.exports.DEFAULT_REFRESH_TIME = 10
module.exports.GRC_REFRESH_INTERVAL_COOKIE = 'grc-refresh-interval-cookie'
/*side panel default auto refresh every 60 minutes*/
module.exports.DEFAULT_SIDE_PANEL_REFRESH_TIME = 3600
module.exports.GRC_SIDE_PANEL_REFRESH_INTERVAL_COOKIE = 'grc-side-panel-refresh-interval-cookie'
module.exports.GRC_VIEW_STATE_COOKIE = 'grc-view-state-cookie'
module.exports.GRC_FILTER_STATE_COOKIE = 'grc-filter-state-cookie'

module.exports.RESOURCE_TYPES = {
  HCM_POLICIES: { name: 'HCMPolicy', list: 'HCMPolicyList' },
  HCM_POLICIES_PER_APPLICATION: { name: 'HCMPolicyApplication', list: 'HCMPolicyApplicationList' },
  HCM_POLICIES_PER_CLUSTER: { name: 'HCMPolicyCluster', list: 'HCMPolicyClusterList' },
  HCM_POLICIES_PER_POLICY: { name: 'HCMPolicyPolicy', list: 'HCMPolicyPolicyList' },
  HCM_COMPLIANCES: { name: 'HCMCompliance', list: 'HCMComplianceList' },
  HCM_CLUSTERS: { name: 'HCMCluster', list: 'HCMClusterList'},// filter: true },
  HCM_SECURITY_FINDINGS: { name: 'HCMSecurityFindings', list: 'HCMSecurityFindingsList' },
  HCM_CLUSTER_FINDINGS: { name: 'HCMClusterFindings', list: 'HCMClusterFindingsList' },
}

module.exports.SECURITY_TYPES = {
  HIGH: 'High',
  MEDIUM: 'Medium',
  LOW: 'Low',
  VIOLATIONS: 'Violations',
}

// keep order from lowest to highest role
module.exports.ROLES = {
  VIEWER: 'Viewer',
  EDITOR: 'Editor',
  OPERATOR: 'Operator',
  ADMIN: 'Administrator',
  ACCOUNT_ADMIN: 'AccountAdministrator',
  CLUSTER_ADMIN: 'ClusterAdministrator'
}
