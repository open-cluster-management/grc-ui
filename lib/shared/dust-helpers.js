/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

const dust = require('dustjs-helpers'),
    config = require('./config'),
    i18n = require('node-i18n-util'),
    path = require('path')

dust.helpers.properties = (chunk, context, bodies, params) => {
  const relativePath = `../../${params.href}` // load the properties

  require(relativePath)

  // resolve the bundle locale being used
  const bundlePath = path.join(__dirname, relativePath),
      locale = i18n._resolveBundle(bundlePath, params.locale).locale,
      basePath = params.href.substring(0, params.href.indexOf('.properties'))

  params.href = basePath + (locale ? '_' + locale.replace('-', '_') : '') + '.js'

  return `${config.contextPath}/${params.href}`
}
