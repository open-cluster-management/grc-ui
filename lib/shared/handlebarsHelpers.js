/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

const config = require('./config'),
      i18n = require('node-i18n-util'),
      path = require('path')

module.exports = function (chunk, context, bodies, params) {
  if (params && params.href) {
    const relativePath = `../../${params.href}` // load the properties

    require(relativePath)

    // resolve the bundle locale being used
    const bundlePath = path.join(__dirname, relativePath),
          locale = i18n._resolveBundle(bundlePath, params.locale).locale,
          basePath = params.href.substring(0, params.href.indexOf('.properties'))

    params.href = basePath + (locale ? '_' + locale.replace('-', '_') : '') + '.js'

    return `${config.contextPath}/${params.href}`
  }
  return config.contextPath
}
