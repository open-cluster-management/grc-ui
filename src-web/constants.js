/* Copyright (c) 2020 Red Hat, Inc. */

export default {
  EMPTY_CHOICE: 'EMPTY_CHOICE',
  overviewTopInfoClustersStr: 'string.clusters',
  overviewTopInfoAppStr: 'string.applications'
}
