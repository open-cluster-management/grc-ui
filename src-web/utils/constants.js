/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

module.exports.ACM_ACCESS_COOKIE = 'acm-access-token-cookie'
module.exports.CFC_AUTH_COOKIE = 'cfc-acs-auth-cookie'

module.exports.REFRESH_TIMES = [5, 10, 30, 60, 5*60, 30*60, 0]
module.exports.DEFAULT_REFRESH_TIME = 10
module.exports.GRC_REFRESH_INTERVAL_COOKIE = 'grc-refresh-interval-cookie'
module.exports.GRC_VIEW_STATE_COOKIE = 'grc-view-state-cookie'
module.exports.GRC_SEARCH_STATE_COOKIE = 'grc-search-state-cookie'
module.exports.GRC_FILTER_STATE_COOKIE = 'grc-filter-state-cookie'

module.exports.RESOURCE_TYPES = {
  POLICIES_BY_CLUSTER: { name: 'PoliciesByCluster', query: 'ALL_POLICIES' },
  POLICIES_BY_POLICY: { name: 'HCMCompliance', query: 'ALL_POLICIES' },
  POLICY: { name: 'Policy', query: 'PoliciesList' },
  PLACEMENT_RULE: { name: 'PlacementRule', query: 'PlacementRulesList' },
  PLACEMENT_BINDING: { name: 'PlacementBinding', query: 'PlacementBindingsList' },
  ANSIBLE_JOB: {name: 'AnsibleJob', query: 'AnsibleJob'}
}

module.exports.INITIAL_POLL_INTERVAL = 10
module.exports.REFRESH_INTERVALS = [5, 10, 30, 60, 5 * 60, 30 * 60, 0]
module.exports.REFRESH_INTERVAL_COOKIE = 'acm-grc-refresh-interval'
