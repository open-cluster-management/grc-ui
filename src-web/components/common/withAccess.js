/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import lodash from 'lodash'
import { connect } from 'react-redux'
import { Redirect } from 'react-router-dom'
import { ROLES } from '../../../lib/shared/constants'
import config from '../../../lib/shared/config'

const withAccess = (ChildComponent, lowestRole) => {
  class checkAccess extends React.PureComponent {
    static propTypes = {
      role: PropTypes.string.isRequired,
    }

    render() {
      const { role } = this.props
      const roleKey = lodash.findKey(ROLES, value => value.toLowerCase() === role.toLowerCase())
      const lowestRoleKey = lodash.findKey(ROLES, value => value.toLowerCase() === lowestRole.toLowerCase())
      return Object.keys(ROLES).indexOf(roleKey) >= Object.keys(ROLES).indexOf(lowestRoleKey) ? <ChildComponent {...this.props} /> : <Redirect to={`${config.contextPath}/welcome`} />
    }
  }


  const mapStateToProps = (state) => {
    const role = state.role && state.role.role
    return { role }
  }

  return connect(mapStateToProps)(checkAccess)
}


export default withAccess
