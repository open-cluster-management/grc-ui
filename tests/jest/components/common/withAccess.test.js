/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import withAccess from '../../../../src-web/components/common/WithAccess'
import { ROLES } from '../../../../lib/shared/constants'
import React from 'react'

describe('withaccess component test', () => {
  it('renders as expected', () => {
    const child = <React.Component></React.Component>
    expect(withAccess(child, ROLES.OPERATOR)).toMatchSnapshot()
  })
})
