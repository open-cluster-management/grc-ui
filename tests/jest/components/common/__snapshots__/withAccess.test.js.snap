// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withaccess component test renders as expected 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "WrappedComponent": [Function],
  "compare": null,
  "displayName": "Connect(CheckAccess)",
  "type": [Function],
}
`;
